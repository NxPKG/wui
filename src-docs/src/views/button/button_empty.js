import React from 'react';

import {
  WuiButtonEmpty,
  WuiFlexGroup,
  WuiFlexItem,
} from '../../../../src/components/';

export default () => (
  <div>
    <WuiFlexGroup gutterSize="s" alignItems="center">
      <WuiFlexItem grow={false}>
        <WuiButtonEmpty onClick={() => window.alert('Button clicked')}>
          Primary
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty size="s" onClick={() => window.alert('Button clicked')}>
          small
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          size="xs"
          onClick={() => window.alert('Button clicked')}>
          extra small
        </WuiButtonEmpty>
      </WuiFlexItem>
    </WuiFlexGroup>

    <WuiFlexGroup gutterSize="s" alignItems="center">
      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown">
          Primary
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          size="s"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown">
          small
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown"
          iconSide="right">
          Primary
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          size="s"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown"
          iconSide="right">
          small
        </WuiButtonEmpty>
      </WuiFlexItem>
    </WuiFlexGroup>

    <WuiFlexGroup gutterSize="s" alignItems="center">
      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          onClick={() => window.alert('Button clicked')}>
          Danger
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          size="s"
          onClick={() => window.alert('Button clicked')}>
          small
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          size="xs"
          onClick={() => window.alert('Button clicked')}>
          extra small
        </WuiButtonEmpty>
      </WuiFlexItem>
    </WuiFlexGroup>

    <WuiFlexGroup gutterSize="s" alignItems="center">
      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown">
          Danger
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          size="s"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown">
          small
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown"
          iconSide="right">
          Danger
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          size="s"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown"
          iconSide="right">
          small
        </WuiButtonEmpty>
      </WuiFlexItem>
    </WuiFlexGroup>

    <WuiFlexGroup gutterSize="s" alignItems="center">
      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="text"
          onClick={() => window.alert('Button clicked')}>
          Text
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="text"
          size="s"
          onClick={() => window.alert('Button clicked')}>
          small
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="text"
          size="xs"
          onClick={() => window.alert('Button clicked')}>
          extra small
        </WuiButtonEmpty>
      </WuiFlexItem>
    </WuiFlexGroup>

    <WuiFlexGroup gutterSize="s" alignItems="center">
      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="text"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown">
          Text
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="text"
          size="s"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown">
          small
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="text"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown"
          iconSide="right">
          Text
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="text"
          size="s"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown"
          iconSide="right">
          small
        </WuiButtonEmpty>
      </WuiFlexItem>
    </WuiFlexGroup>

    <WuiFlexGroup gutterSize="s" alignItems="center">
      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          onClick={() => window.alert('Button clicked')}
          isDisabled>
          Disabled
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          size="s"
          onClick={() => window.alert('Button clicked')}
          isDisabled>
          small
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          size="xs"
          onClick={() => window.alert('Button clicked')}
          isDisabled>
          extra small
        </WuiButtonEmpty>
      </WuiFlexItem>
    </WuiFlexGroup>

    <WuiFlexGroup gutterSize="s" alignItems="center">
      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          onClick={() => window.alert('Button clicked')}
          isLoading>
          Loading
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          onClick={() => window.alert('Button clicked')}
          isLoading
          iconSide="right">
          Loading
        </WuiButtonEmpty>
      </WuiFlexItem>
    </WuiFlexGroup>

    <WuiFlexGroup gutterSize="s" alignItems="center">
      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown"
          isDisabled>
          Disabled
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          size="s"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown"
          isDisabled>
          small
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown"
          iconSide="right"
          isDisabled>
          Disabled
        </WuiButtonEmpty>
      </WuiFlexItem>

      <WuiFlexItem grow={false}>
        <WuiButtonEmpty
          color="danger"
          size="s"
          onClick={() => window.alert('Button clicked')}
          iconType="arrowDown"
          iconSide="right"
          isDisabled>
          small
        </WuiButtonEmpty>
      </WuiFlexItem>
    </WuiFlexGroup>
  </div>
);
