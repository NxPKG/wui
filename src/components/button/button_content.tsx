/*
 * Copyright 2022 Wazuh Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * NOTICE: THIS FILE HAS BEEN MODIFIED BY WAZUH INC UNDER COMPLIANCE WITH THE APACHE 2.0 LICENSE FROM THE ORIGINAL WORK
 * OF THE COMPANY Elasticsearch B.V.
 *
 * THE FOLLOWING IS THE COPYRIGHT OF THE ORIGINAL DOCUMENT:
 *
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, { HTMLAttributes, FunctionComponent } from 'react';
import classNames from 'classnames';
import { CommonProps, keysOf } from '../common';
import { WuiLoadingSpinner } from '../loading';
import { WuiIcon, IconType } from '../icon';

export type ButtonContentIconSide = 'left' | 'right';

const iconSideToClassNameMap: {
  [side in ButtonContentIconSide]: string | null;
} = {
  left: null,
  right: 'wuiButtonContent--iconRight',
};

export const ICON_SIDES = keysOf(iconSideToClassNameMap);

export type WuiButtonContentType = HTMLAttributes<HTMLSpanElement>;

/**
 * *INTERNAL ONLY*
 * This component is simply a helper component for reuse within other button components
 */
export interface WuiButtonContentProps extends CommonProps {
  iconType?: IconType;
  iconSide?: ButtonContentIconSide;
  isLoading?: boolean;
  /**
   * Object of props passed to the <span/> wrapping the content's text/children only (not icon)
   */
  textProps?: HTMLAttributes<HTMLSpanElement> & CommonProps;
}

export const WuiButtonContent: FunctionComponent<WuiButtonContentType &
  WuiButtonContentProps> = ({
  children,
  textProps,
  isLoading = false,
  iconType,
  iconSide = 'left',
  ...contentProps
}) => {
  // Add an icon to the button if one exists.
  let buttonIcon;

  if (isLoading) {
    buttonIcon = (
      <WuiLoadingSpinner className="wuiButtonContent__spinner" size="m" />
    );
  } else if (iconType) {
    buttonIcon = (
      <WuiIcon className="wuiButtonContent__icon" type={iconType} size="m" />
    );
  }

  const contentClassNames = classNames(
    'wuiButtonContent',
    iconSide ? iconSideToClassNameMap[iconSide] : null,
    contentProps && contentProps.className
  );

  return (
    <span {...contentProps} className={contentClassNames}>
      {buttonIcon}
      <span {...textProps}>{children}</span>
    </span>
  );
};
