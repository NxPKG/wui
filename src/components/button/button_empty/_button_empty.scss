/**
 * 1. We don't want any of the animations that come inherited from the mixin.
 *    These should act like normal links instead.
 * 2. Change the easing, quickness to not bounce so lighter backgrounds don't flash
 */
.wuiButtonEmpty {
  @include wuiButton;

  border-color: transparent;
  background-color: transparent;
  box-shadow: none;
  // sass-lint:disable-block no-important
  transform: none !important; /* 1 */
  animation: none !important; /* 1 */
  transition-timing-function: ease-in; /* 2 */
  transition-duration: $wuiAnimSpeedFast; /* 2 */

  .wuiButtonEmpty__content {
    padding: 0 $wuiSizeS;
  }

  .wuiButtonEmpty__text {
    text-overflow: ellipsis;
    overflow: hidden;
  }

  &.wuiButtonEmpty--small {
    height: $wuiButtonHeightSmall;
  }

  &.wuiButtonEmpty--xSmall {
    height: $wuiSizeL;
    font-size: $wuiFontSizeS;
  }

  &:disabled {
    @include wuiButtonContentDisabled;

    color: $wuiButtonColorDisabledText;

    &:focus {
      background-color: transparent;
    }

    &:hover,
    &:focus {
      text-decoration: none;
    }
  }
}


.wuiButtonEmpty--flushLeft,
.wuiButtonEmpty--flushRight,
.wuiButtonEmpty--flushBoth {
  .wuiButtonEmpty__content {
    padding-left: 0;
    padding-right: 0;
  }
}

.wuiButtonEmpty--flushLeft {
  margin-right: $wuiSizeS;
}

.wuiButtonEmpty--flushRight {
  margin-left: $wuiSizeS;
}

// Modifier naming and colors.
$wuiButtonEmptyTypes: (
  primary: $wuiColorPrimary,
  danger: $wuiColorDanger,
  disabled: $wuiButtonColorDisabledText,
  ghost: $wuiColorGhost,
  text: $wuiTextColor,
);

// Create button modifiers based upon the map.
@each $name, $color in $wuiButtonEmptyTypes {
  .wuiButtonEmpty--#{$name} {
    @if ($name == 'ghost') {
      // Ghost is unique and ALWAYS sits against a dark background.
      color: $color;
    } @else if ($name == 'text') {
      // The default color is lighter than the normal text color, make the it the text color
      color: $wuiTextColor;
    } @else {
      // Other colors need to check their contrast against the page background color.
      color: makeHighContrastColor($color, $wuiColorEmptyShade);
    }

    &:focus {
      background-color: transparentize($color, .9);
    }

    &:hover {
      @if ($name == 'disabled') {
        cursor: not-allowed;
      }
    }
  }
}

