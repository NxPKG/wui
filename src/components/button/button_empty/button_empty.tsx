/*
 * Copyright 2022 Wazuh Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * NOTICE: THIS FILE HAS BEEN MODIFIED BY WAZUH INC UNDER COMPLIANCE WITH THE APACHE 2.0 LICENSE FROM THE ORIGINAL WORK
 * OF THE COMPANY Elasticsearch B.V.
 *
 * THE FOLLOWING IS THE COPYRIGHT OF THE ORIGINAL DOCUMENT:
 *
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

import {
  CommonProps,
  ExclusiveUnion,
  PropsForAnchor,
  PropsForButton,
  keysOf,
} from '../../common';
import { getSecureRelForTarget } from '../../../services';
import {
  WuiButtonContent,
  WuiButtonContentProps,
  WuiButtonContentType,
} from '../button_content';
import { validateHref } from '../../../services/security/href_validator';

export type WuiButtonEmptyColor =
  | 'primary'
  | 'danger'
  | 'disabled'
  | 'text'
  | 'ghost';

const colorToClassNameMap: { [color in WuiButtonEmptyColor]: string } = {
  primary: 'wuiButtonEmpty--primary',
  danger: 'wuiButtonEmpty--danger',
  disabled: 'wuiButtonEmpty--disabled',
  text: 'wuiButtonEmpty--text',
  ghost: 'wuiButtonEmpty--ghost',
};

export const COLORS = keysOf(colorToClassNameMap);

const sizeToClassNameMap = {
  xs: 'wuiButtonEmpty--xSmall',
  s: 'wuiButtonEmpty--small',
  l: 'wuiButtonEmpty--large',
};

export const SIZES = keysOf(sizeToClassNameMap);

export type WuiButtonEmptySizes = keyof typeof sizeToClassNameMap;

const flushTypeToClassNameMap = {
  left: 'wuiButtonEmpty--flushLeft',
  right: 'wuiButtonEmpty--flushRight',
  both: 'wuiButtonEmpty--flushBoth',
};

export const FLUSH_TYPES = keysOf(flushTypeToClassNameMap);

/**
 * Extends WuiButtonContentProps which provides
 * `iconType`, `iconSide`, and `textProps`
 */
interface CommonWuiButtonEmptyProps extends WuiButtonContentProps, CommonProps {
  /**
   * Any of our named colors
   */
  color?: WuiButtonEmptyColor;
  size?: WuiButtonEmptySizes;
  /**
   * Ensure the text of the button sits flush to the left, right, or both sides of its container
   */
  flush?: keyof typeof flushTypeToClassNameMap;
  /**
   * `disabled` is also allowed
   */
  isDisabled?: boolean;
  /**
   * Force disables the button and changes the icon to a loading spinner
   */
  isLoading?: boolean;
  href?: string;
  target?: string;
  rel?: string;
  type?: 'button' | 'submit';
  buttonRef?: (ref: HTMLButtonElement | HTMLAnchorElement | null) => void;
  /**
   * Object of props passed to the <span/> wrapping the button's content
   */
  contentProps?: WuiButtonContentType;
}

type WuiButtonEmptyPropsForAnchor = PropsForAnchor<CommonWuiButtonEmptyProps>;

type WuiButtonEmptyPropsForButton = PropsForButton<CommonWuiButtonEmptyProps>;

export type WuiButtonEmptyProps = ExclusiveUnion<
  WuiButtonEmptyPropsForAnchor,
  WuiButtonEmptyPropsForButton
>;

export const WuiButtonEmpty: FunctionComponent<WuiButtonEmptyProps> = ({
  children,
  className,
  iconType,
  iconSide = 'left',
  color = 'primary',
  size,
  flush,
  isDisabled: _isDisabled,
  disabled: _disabled,
  isLoading,
  href,
  target,
  rel,
  type = 'button',
  buttonRef,
  contentProps,
  textProps,
  ...rest
}) => {
  const isHrefValid = !href || validateHref(href);
  const disabled = _disabled || !isHrefValid;
  const isDisabled = _isDisabled || !isHrefValid;

  // If in the loading state, force disabled to true
  const buttonIsDisabled = isLoading || isDisabled || disabled;

  const classes = classNames(
    'wuiButtonEmpty',
    colorToClassNameMap[color],
    size ? sizeToClassNameMap[size] : null,
    flush ? flushTypeToClassNameMap[flush] : null,
    {
      'wuiButtonEmpty-isDisabled': buttonIsDisabled,
    },
    className
  );

  const contentClassNames = classNames(
    'wuiButtonEmpty__content',
    contentProps && contentProps.className
  );

  const textClassNames = classNames(
    'wuiButtonEmpty__text',
    textProps && textProps.className
  );

  const innerNode = (
    <WuiButtonContent
      isLoading={isLoading}
      iconType={iconType}
      iconSide={iconSide}
      textProps={{ ...textProps, className: textClassNames }}
      {...contentProps}
      // className has to come last to override contentProps.className
      className={contentClassNames}>
      {children}
    </WuiButtonContent>
  );

  // <a> elements don't respect the `disabled` attribute. So if we're disabled, we'll just pretend
  // this is a button and piggyback off its disabled styles.
  if (href && !buttonIsDisabled) {
    const secureRel = getSecureRelForTarget({ href, target, rel });

    return (
      <a
        className={classes}
        href={href}
        target={target}
        rel={secureRel}
        ref={buttonRef}
        {...(rest as WuiButtonEmptyPropsForAnchor)}>
        {innerNode}
      </a>
    );
  }

  return (
    <button
      disabled={buttonIsDisabled}
      className={classes}
      type={type}
      ref={buttonRef}
      {...(rest as WuiButtonEmptyPropsForButton)}>
      {innerNode}
    </button>
  );
};
