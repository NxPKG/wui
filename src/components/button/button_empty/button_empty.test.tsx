/*
 * Copyright 2022 Wazuh Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * NOTICE: THIS FILE HAS BEEN MODIFIED BY WAZUH INC UNDER COMPLIANCE WITH THE APACHE 2.0 LICENSE FROM THE ORIGINAL WORK
 * OF THE COMPANY Elasticsearch B.V.
 *
 * THE FOLLOWING IS THE COPYRIGHT OF THE ORIGINAL DOCUMENT:
 *
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { render, mount } from 'enzyme';
import { requiredProps } from '../../../test/required_props';

import { WuiButtonEmpty, COLORS, SIZES, FLUSH_TYPES } from './button_empty';
import { ICON_SIDES } from '../button_content';

describe('WuiButtonEmpty', () => {
  test('is rendered', () => {
    const component = render(
      <WuiButtonEmpty {...requiredProps}>Content</WuiButtonEmpty>
    );

    expect(component).toMatchSnapshot();
  });

  describe('props', () => {
    describe('isDisabled', () => {
      it('is rendered', () => {
        const component = render(<WuiButtonEmpty isDisabled />);

        expect(component).toMatchSnapshot();
      });

      it('renders a button even when href is defined', () => {
        const component = render(<WuiButtonEmpty href="#" isDisabled />);

        expect(component).toMatchSnapshot();
      });

      it('renders if passed simply as disabled', () => {
        const component = render(<WuiButtonEmpty disabled />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('isLoading', () => {
      it('is rendered', () => {
        const component = render(<WuiButtonEmpty isLoading />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('iconType', () => {
      it('is rendered', () => {
        const component = render(<WuiButtonEmpty iconType="user" />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('color', () => {
      COLORS.forEach(color => {
        test(`${color} is rendered`, () => {
          const component = render(<WuiButtonEmpty color={color} />);

          expect(component).toMatchSnapshot();
        });
      });
    });

    describe('size', () => {
      SIZES.forEach(size => {
        test(`${size} is rendered`, () => {
          const component = render(<WuiButtonEmpty size={size} />);

          expect(component).toMatchSnapshot();
        });
      });
    });

    describe('iconSide', () => {
      ICON_SIDES.forEach(iconSide => {
        test(`${iconSide} is rendered`, () => {
          const component = render(
            <WuiButtonEmpty iconType="user" iconSide={iconSide}>
              Content
            </WuiButtonEmpty>
          );

          expect(component).toMatchSnapshot();
        });
      });
    });

    describe('flush', () => {
      FLUSH_TYPES.forEach(flushType => {
        test(`${flushType} is rendered`, () => {
          const component = render(<WuiButtonEmpty flush={flushType} />);

          expect(component).toMatchSnapshot();
        });
      });
    });

    describe('href', () => {
      it('secures the rel attribute when the target is _blank', () => {
        const component = render(<WuiButtonEmpty href="#" target="_blank" />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('onClick', () => {
      it('supports onClick and href', () => {
        const handler = jest.fn();
        const component = mount(<WuiButtonEmpty href="#" onClick={handler} />);
        component.find('a').simulate('click');
        expect(handler.mock.calls.length).toEqual(1);
      });

      it('supports onClick as a button', () => {
        const handler = jest.fn();
        const component = mount(<WuiButtonEmpty onClick={handler} />);
        component.find('button').simulate('click');
        expect(handler.mock.calls.length).toEqual(1);
      });
    });

    test('contentProps is rendered', () => {
      const component = render(
        <WuiButtonEmpty contentProps={requiredProps}>Content</WuiButtonEmpty>
      );

      expect(component).toMatchSnapshot();
    });

    test('textProps is rendered', () => {
      const component = render(
        <WuiButtonEmpty textProps={requiredProps}>Content</WuiButtonEmpty>
      );

      expect(component).toMatchSnapshot();
    });
  });
});
