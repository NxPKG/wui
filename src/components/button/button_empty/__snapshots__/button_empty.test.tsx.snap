// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WuiButtonEmpty is rendered 1`] = `
<button
  aria-label="aria-label"
  class="wuiButtonEmpty wuiButtonEmpty--primary testClass1 testClass2"
  data-test-subj="test subject string"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`WuiButtonEmpty props color danger is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--danger"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props color disabled is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--disabled"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props color ghost is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--ghost"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props color primary is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props color text is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--text"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props contentProps is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary"
  type="button"
>
  <span
    aria-label="aria-label"
    class="wuiButtonContent wuiButtonEmpty__content testClass1 testClass2"
    data-test-subj="test subject string"
  >
    <span
      class="wuiButtonEmpty__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`WuiButtonEmpty props flush both is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary wuiButtonEmpty--flushBoth"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props flush left is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary wuiButtonEmpty--flushLeft"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props flush right is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary wuiButtonEmpty--flushRight"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props href secures the rel attribute when the target is _blank 1`] = `
<a
  class="wuiButtonEmpty wuiButtonEmpty--primary"
  href="#"
  rel="noopener noreferrer"
  target="_blank"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</a>
`;

exports[`WuiButtonEmpty props iconSide left is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <div
      class="wuiButtonContent__icon"
      data-wuiicon-type="user"
    />
    <span
      class="wuiButtonEmpty__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`WuiButtonEmpty props iconSide right is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonContent--iconRight wuiButtonEmpty__content"
  >
    <div
      class="wuiButtonContent__icon"
      data-wuiicon-type="user"
    />
    <span
      class="wuiButtonEmpty__text"
    >
      Content
    </span>
  </span>
</button>
`;

exports[`WuiButtonEmpty props iconType is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <div
      class="wuiButtonContent__icon"
      data-wuiicon-type="user"
    />
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props isDisabled is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary wuiButtonEmpty-isDisabled"
  disabled=""
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props isDisabled renders a button even when href is defined 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary wuiButtonEmpty-isDisabled"
  disabled=""
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props isDisabled renders if passed simply as disabled 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary wuiButtonEmpty-isDisabled"
  disabled=""
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props isLoading is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary wuiButtonEmpty-isDisabled"
  disabled=""
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiLoadingSpinner wuiLoadingSpinner--medium wuiButtonContent__spinner"
    />
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props size l is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary wuiButtonEmpty--large"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props size s is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary wuiButtonEmpty--small"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props size xs is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary wuiButtonEmpty--xSmall"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      class="wuiButtonEmpty__text"
    />
  </span>
</button>
`;

exports[`WuiButtonEmpty props textProps is rendered 1`] = `
<button
  class="wuiButtonEmpty wuiButtonEmpty--primary"
  type="button"
>
  <span
    class="wuiButtonContent wuiButtonEmpty__content"
  >
    <span
      aria-label="aria-label"
      class="wuiButtonEmpty__text testClass1 testClass2"
      data-test-subj="test subject string"
    >
      Content
    </span>
  </span>
</button>
`;
